### ISSUE TITLE
Allow renderIntoDocument on TagOf

### ISSUE THREAD

**Issue by ramnivas** [japgolly/scalajs-react#387]
I have the following piece of code that I expected to work (porting existing code over to neo).

```scala
val myElement: Element = ...
val myDiv = div()(
  ...
)
myDiv.renderIntoDOM(myDiv, myElement)
```

But `renderIntoDOM` is available only on `Unmounted`. Is there a reason `TagOf`s cannot have the same method following React's `[ReactDOM.render](https://facebook.github.io/react/docs/react-dom.html#render)`

**Comment by japgolly** [comment]
I think you're right. Feel free to PR.

Workaround for now: `ReactDOM.raw.render(tag.render.rawElement, element)`

**Comment by ramnivas** [comment]
Thanks. Will PR soon.

**Comment by japgolly** [comment]
Fixed via 58b77d9baed8d0a625393fda9bf4ac28458a8a69
